src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.cpp

```diff
@@ -0,0 +1,645 @@
+        {
+            lastID = candidateID;
+            return candidateID;
+        }
+    }
+
+    return kMaxConnectionId; // All 0 to 65534 IDs are in use
+}
+
+void PushAvStreamTransportServer::HandleAllocatePushTransport(HandlerContext & ctx,
+                                                              const Commands::AllocatePushTransport::DecodableType & commandData)
+{
+    Commands::AllocatePushTransportResponse::Type response;
+    auto & transportOptions = commandData.transportOptions;
+
+    // Todo: TLSEndpointID Validation
+
+    bool isFormatSupported                  = false;
+    IngestMethodsEnum ingestMethod          = commandData.transportOptions.ingestMethod;
+    ContainerOptionsStruct containerOptions = commandData.transportOptions.containerOptions;
+
+    for (auto & supportsFormat : mSupportedFormats)
+    {
+        if ((supportsFormat.ingestMethod == ingestMethod) && (supportsFormat.containerFormat == containerOptions.containerType))
+        {
+            isFormatSupported = true;
+        }
+    }
+
+    if (isFormatSupported == false)
+    {
+        auto status = static_cast<uint8_t>(StatusCodeEnum::kInvalidCombination);
+        ChipLogError(Zcl, "HandleAllocatePushTransport: Invalid Format Combination");
+        ctx.mCommandHandler.AddClusterSpecificFailure(ctx.mRequestPath, status);
+        return;
+    }
+
+    bool isValidUrl = mDelegate.validateUrl(std::string(transportOptions.url.data(), transportOptions.url.size()));
+
+    if (isValidUrl == false)
+    {
+        auto status = static_cast<uint8_t>(StatusCodeEnum::kInvalidURL);
+        ChipLogError(Zcl, "HandleAllocatePushTransport: Invalid Url");
+        ctx.mCommandHandler.AddClusterSpecificFailure(ctx.mRequestPath, status);
+        return;
+    }
+
+    if (transportOptions.triggerOptions.triggerType == TransportTriggerTypeEnum::kUnknownEnumValue)
+    {
+        auto status = static_cast<uint8_t>(StatusCodeEnum::kInvalidTriggerType);
+        ChipLogError(Zcl, "HandleAllocatePushTransport: Invalid Trigger type");
+        ctx.mCommandHandler.AddClusterSpecificFailure(ctx.mRequestPath, status);
+        return;
+    }
+
+    // Todo: Validate MotionZones list in the TransportTriggerOptionsStruct field
+    // Validate Bandwidth Requirement
+    CHIP_ERROR err = mDelegate.ValidateBandwidthLimit(transportOptions.streamUsage, transportOptions.videoStreamID,
+                                                      transportOptions.audioStreamID);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "HandleAllocatePushTransport: Resource Exhausted");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ResourceExhausted);
+        return;
+    }
+
+    // Validate the StreamUsageEnum as per resource management and stream priorities.
+    err =
+        mDelegate.ValidateStreamUsage(transportOptions.streamUsage, transportOptions.videoStreamID, transportOptions.audioStreamID);
```

Comment:
Spec says the selected IDs get put in the transportOptions, but that's clearly impossible for this delegate API.  I don't understand how this matches spec.

In general, it feels like a different API shape is needed here for the delegate: checking whether there is an allocated stream, and/or selecting streams....